File: frontend/folhafacil/src/app/services/hora-extra.service.ts
import { HttpClient } from "@angular/common/http";
import { inject, Injectable } from "@angular/core";
import { environment } from "../../environment/environment";
import { HoraExtraDTO, HoraExtraFilterDTO, HoraExtraResponseDTO } from "../models/horaextra.model";
import { Observable } from "rxjs";

@Injectable({
	providedIn: "root",
})

export class HoraExtraService{
	http = inject(HttpClient);

    url = `${environment.API_URL}hora-extra`

    inicar(i : HoraExtraDTO){
        return this.http.post(`${this.url}/iniciar`, {...i});
    }

    buscar(f: HoraExtraFilterDTO): Observable<HoraExtraResponseDTO[]>{
       return this.http.post<HoraExtraResponseDTO[]>(`${this.url}/buscar`, {...f})
    }

    minhasHoras() : Observable<HoraExtraResponseDTO[]>{
        return this.http.get<HoraExtraResponseDTO[]>(`${this.url}/minhas`)
    }

    finalizar(id: number){
        return this.http.get(`${this.url}/${id}/finalizar`)
    }

    cancelar(id: number){
        return this.http.get(`${this.url}/${id}/cancelar`)
    }
}

File: backend/folhafacil/src/main/java/com/folhafacil/folhafacil/service/HoraExtra/HoraExtraService.java
package com.folhafacil.folhafacil.service.HoraExtra;

import com.folhafacil.folhafacil.dto.HoraExtra.HoraExtraDTO;
import com.folhafacil.folhafacil.dto.HoraExtra.HoraExtraFilterDTO;
import com.folhafacil.folhafacil.dto.HoraExtra.HoraExtraReponseDTO;
import org.springframework.security.oauth2.jwt.Jwt;

import java.util.List;

public interface HoraExtraService {
    public void iniciar(HoraExtraDTO d, Jwt token) throws RuntimeException;

    public void finalizar(Long id) throws RuntimeException;

    public void cancelar(Long id) throws RuntimeException;

    public List<HoraExtraReponseDTO> buscar(HoraExtraFilterDTO f);

    public List<HoraExtraReponseDTO> minhasHoras(Jwt jwt);
}
